src/hotspot/cpu/riscv/stubGenerator_riscv.cpp

```diff
@@ -3659,8 +3659,417 @@ class StubGenerator: public StubCodeGenerator {
       return entry;
     }
   };
+
 #endif // COMPILER2
 
+#undef __
+#define __ this->
+  class Sha2Generator : public MacroAssembler {
+    StubCodeGenerator* _cgen;
+   public:
+      Sha2Generator(MacroAssembler* masm, StubCodeGenerator* cgen) : MacroAssembler(masm->code()), _cgen(cgen) {}
+      address generate_sha256_implCompress(bool multi_block) {
+        return generate_sha2_implCompress<Assembler::e32>(multi_block);
+      }
+      address generate_sha512_implCompress(bool multi_block) {
+        return generate_sha2_implCompress<Assembler::e64>(multi_block);
+      }
+   private:
+
+    template<Assembler::SEW T>
+    void vl1reXX_v(VectorRegister vr, Register sr) {
+      if (T == Assembler::e32) __ vl1re32_v(vr, sr);
```

Comment:
related comments as above about the template usage, seems here it's more suitable to use a parameter.
Maybe also in some other places below.